### PR TITLE
Migrate pipeline errors using pagination

### LINKED ISSUE #2737: Problems with DB migration `convert-to-new-pipeline-error-format`
### Component

server

### Describe the bug

The server startup stuck on `convert-to-new-pipeline-error-format` migration. Even after 120m the migration has not finished.

```json
{"level":"info","time":"2023-11-03T12:05:16Z","caller":"/build/cmd/common/logger.go:100","message":"LogLevel = debug"}
{"level":"info","time":"2023-11-03T12:05:36Z","caller":"/build/server/store/datastore/migration/migration.go:245","message":"Migration 'convert-to-new-pipeline-error-format' is still running, please be patient"}
{"level":"info","time":"2023-11-03T12:05:56Z","caller":"/build/server/store/datastore/migration/migration.go:245","message":"Migration 'convert-to-new-pipeline-error-format' is still running, please be patient"}
```

After enabling the XORM log in woodpecker the log shows the executed SQL commands as well:
<details>
  <summary>Server Logs</summary>

```json
{
  "level": "info",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/common/logger.go:100",
  "message": "LogLevel = debug"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT tablename FROM pg_tables WHERE schemaname = $1 [public] - 30.308275ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT column_name, column_default, is_nullable, data_type, character_maximum_length, description,\n    CASE WHEN p.contype = 'p' THEN true ELSE false END AS primarykey,\n    CASE WHEN p.contype = 'u' THEN true ELSE false END AS uniquekey\nFROM pg_attribute f\n    JOIN pg_class c ON c.oid = f.attrelid JOIN pg_type t ON t.oid = f.atttypid\n    LEFT JOIN pg_attrdef d ON d.adrelid = c.oid AND d.adnum = f.attnum\n    LEFT JOIN pg_description de ON f.attrelid=de.objoid AND f.attnum=de.objsubid\n    LEFT JOIN pg_namespace n ON n.oid = c.relnamespace\n    LEFT JOIN pg_constraint p ON p.conrelid = c.oid AND f.attnum = ANY (p.conkey)\n    LEFT JOIN pg_class AS g ON p.confrelid = g.oid\n    LEFT JOIN INFORMATION_SCHEMA.COLUMNS s ON s.column_name=f.attname AND c.relname=s.table_name\nWHERE n.nspname= s.table_schema AND c.relkind = 'r' AND c.relname = $1 AND s.table_schema = $2 AND f.attnum > 0 ORDER BY f.attnum; [migrations public] - 28.791397ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT indexname, indexdef FROM pg_indexes WHERE tablename=$1 AND schemaname=$2 [migrations public] - 11.22641ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] BEGIN TRANSACTION [] - 603.478µs"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT count(*) FROM \"migrations\" [] - 1.181009ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] COMMIT [] - 1.832027ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT \"name\" FROM \"migrations\" [] - 1.039455ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] BEGIN TRANSACTION [] - 928.707µs"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT tablename FROM pg_tables WHERE schemaname = $1 [public] - 2.278934ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT column_name, column_default, is_nullable, data_type, character_maximum_length, description,\n    CASE WHEN p.contype = 'p' THEN true ELSE false END AS primarykey,\n    CASE WHEN p.contype = 'u' THEN true ELSE false END AS uniquekey\nFROM pg_attribute f\n    JOIN pg_class c ON c.oid = f.attrelid JOIN pg_type t ON t.oid = f.atttypid\n    LEFT JOIN pg_attrdef d ON d.adrelid = c.oid AND d.adnum = f.attnum\n    LEFT JOIN pg_description de ON f.attrelid=de.objoid AND f.attnum=de.objsubid\n    LEFT JOIN pg_namespace n ON n.oid = c.relnamespace\n    LEFT JOIN pg_constraint p ON p.conrelid = c.oid AND f.attnum = ANY (p.conkey)\n    LEFT JOIN pg_class AS g ON p.confrelid = g.oid\n    LEFT JOIN INFORMATION_SCHEMA.COLUMNS s ON s.column_name=f.attname AND c.relname=s.table_name\nWHERE n.nspname= s.table_schema AND c.relkind = 'r' AND c.relname = $1 AND s.table_schema = $2 AND f.attnum > 0 ORDER BY f.attnum; [pipelines public] - 74.527942ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT indexname, indexdef FROM pg_indexes WHERE tablename=$1 AND schemaname=$2 [pipelines public] - 5.04695ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] DROP INDEX \"public\".\"IDX_pipelines_pipeline_author\" [] - 6.259148ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] DROP INDEX \"public\".\"IDX_pipelines_pipeline_repo_id\" [] - 1.170448ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] DROP INDEX \"public\".\"IDX_pipelines_pipeline_status\" [] - 1.039996ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] DROP INDEX \"public\".\"UQE_pipelines_s\" [] - 1.358872ms"
}
{
  "level": "info",
  "component": "xorm",
  "time": "2023-11-03T22:40:20Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/xorm.go:77",
  "message": "[SQL] SELECT tablename FROM pg_tables WHERE schemaname = $1 [public] - 1.928387ms"
}
{
  "level": "info",
  "time": "2023-11-03T22:40:40Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/migration/migration.go:245",
  "message": "Migration 'convert-to-new-pipeline-error-format' is still running, please be patient"
}
{
  "level": "info",
  "time": "2023-11-03T22:41:00Z",
  "caller": "/woodpecker/src/github.com/woodpecker-ci/woodpecker/server/store/datastore/migration/migration.go:245",
  "message": "Migration 'convert-to-new-pipeline-error-format' is still running, please be patient"
}
```
</details>

I was also looking into `pg_stat_activity` to find long-running queries:

```
------------+------------------------------------------------------------------------------------------------------------------------------------
       3914 | SELECT column_name, column_default, is_nullable, data_type, character_maximum_length, description,                                +
            |     CASE WHEN p.contype = 'p' THEN true ELSE false END AS primarykey,                                                             +
            |     CASE WHEN p.contype = 'u' THEN true ELSE false END AS uniquekey                                                               +
            | FROM pg_attribute f                                                                                                               +
            |     JOIN pg_class c ON c.oid = f.attrelid JOIN pg_type t ON t.oid = f.atttypid                                                    +
            |     LEFT JOIN pg_attrdef d ON d.adrelid = c.oid AND d.adnum = f.attnum                                                            +
            |     LEFT JOIN pg_description de ON f.attrelid=de.objoid AND f.attnum=de.objsubid                                                  +
            |     LEFT JOIN pg_namespace n ON n.oid = c.relnamespace                                                                            +
            |     LEFT JOIN pg_constraint p ON p.conrelid = c.oid AND f.attnum = ANY (p.conkey)                                                 +
            |     LEFT JOIN pg_class AS g ON p.confrelid = g.oid                                                                                +
            |     LEFT JOIN INFORMATION_SCHEMA.COLUMNS s ON s.column_name=f.attname AND c.relname=s.table_name                                  +
            | WHERE n.nspname= s.table_schema AND c.relkind = 'r' AND c.relname = $1 AND s.table_schema = $2 AND f.attnum > 0 ORDER BY f.attnum;
```

```
       age       | state  | process_id |                                                            active_query                                                            
-----------------+--------+------------+------------------------------------------------------------------------------------------------------------------------------------
 00:17:46.638525 | active |       3914 | SELECT column_name, column_default, is_nullable, data_type, character_maximum_length, description,                                +
                 |        |            |     CASE WHEN p.contype = 'p' THEN true ELSE false END AS primarykey,                                                             +
                 |        |            |     CASE WHEN p.contype = 'u' THEN true ELSE false END AS uniquekey                                                               +
                 |        |            | FROM pg_attribute f                                                                                                               +
                 |        |            |     JOIN pg_class c ON c.oid = f.attrelid JOIN pg_type t ON t.oid = f.atttypid                                                    +
```

I've not found any errors in the general log of the postgres server.

### System Info

```shell
`next-dffdcdaac0`
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

### PR DESCRIPTION
**Pull request by anbraten**
closes #2737

### CODE REVIEW

**Comment by anbraten** [comment]
> Does this fix https://github.com/woodpecker-ci/woodpecker/issues/2737?

I hope so, but wanted to create test images first for the reporting users to check. On my deployment it worked from the beginning on.

File: server/store/datastore/migration/026_convert_to_new_pipeline_errors_format.go [hunk 1]
```diff
@@ -15,11 +15,16 @@
 package migration
```
Comment (anbraten) [inline server/store/datastore/migration/026_convert_to_new_pipeline_errors_format.go:15]: Waiting for user feedback ...

Comment (pat-s) [reply server/store/datastore/migration/026_convert_to_new_pipeline_errors_format.go:15]: I can try this PR on my k8s install and report back?

File: server/store/datastore/migration/026_convert_to_new_pipeline_errors_format.go [hunk 2]
```diff
@@ -59,25 +64,36 @@ var convertToNewPipelineErrorFormat = task{
 			return err
```
Comment (anbraten) [inline server/store/datastore/migration/026_convert_to_new_pipeline_errors_format.go:64]: Okay error is actually coming from using `.Sync` twice in the migration / session.

**Pull request merged**